assignment/assignment-tool/tool/src/test/org/sakaiproject/assignment/tool/AssignmentActionTestTools.java

```diff
@@ -0,0 +1,132 @@
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sakaiproject.assignment.api.AssignmentConstants;
+import org.sakaiproject.assignment.api.AssignmentService;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.component.cover.ServerConfigurationService;
+import org.sakaiproject.event.api.SessionState;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.Tool;
+import org.sakaiproject.tool.cover.ToolManager;
+import org.sakaiproject.util.FormattedText;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.*;
+
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests for AssignmentAction
+ */
+@SuppressWarnings("deprecation")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ComponentManager.class})
+@Slf4j
+public class AssignmentActionTestTools { 
+
+    private AssignmentAction assignmentAction;
+    @Mock
+    private AssignmentService assignmentService;
+
+    @Before
+    public void setUp() {
+        BasicConfigurator.configure();
+        PowerMockito.mockStatic(ComponentManager.class);
+        // A mock component manager.
+        when(ComponentManager.get(any(Class.class))).then(new Answer<Object>() {
+            private Map<Class, Object> mocks = new HashMap<>();
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Class classToMock = (Class) invocation.getArguments()[0];
+                return mocks.computeIfAbsent(classToMock, k -> mock(classToMock));
+            }
+        });
+        
+        when(ComponentManager.get(SessionManager.class).getCurrentSession()).thenReturn(mock(Session.class));
+        when(FormattedText.getDecimalSeparator()).thenReturn(".");
+        
+        when(FormattedText.getNumberFormat()).thenReturn(NumberFormat.getInstance(Locale.ENGLISH));
+        assignmentAction = new AssignmentAction();
+
+        Mockito.when(ComponentManager.get(AssignmentService.class)).thenReturn(assignmentService);
+
+    }
+	
+    //This probably should also be moved from AssignmentAction to a util or something
+    public Integer getScaleFactor(Integer decimals) {		
+        return (int)Math.pow(10.0, decimals);
+    }
+
+    @Test
+    public void testScalePointGrade() {
+        SessionState state = mock(SessionState.class);
+        //Simple state?
+        final Map <String,Object> stateAttribute = new HashMap<String,Object> ();
+        // mock for method setAttributepersist
+        doAnswer(new Answer<Void>() {
```

Comment:
Yeah I thought about that, as this is basically what this is doing here. We really need a more working mock. Either a stub, fake, whatever it's called! I think it would be more readable if anything else ever needed it for sure.